hardware/hardware.mk

```diff
@@ -144,6 +144,23 @@ $(HARDWARE_PATH)/%.v: $(HARDWARE_PATH)/%.rom
 	env python2 $(BASEJUMP_STL_DIR)/bsg_mem/bsg_ascii_to_rom.py $< \
                bsg_bladerunner_configuration > $@
 
+
+# the default case for memory config features
+CL_MANYCORE_MEMSYS_FEATURE_DMA   := 0
+CL_MANYCORE_MEMSYS_FEATURE_CACHE := 1
+
+# do we use dramsim3?
+ifneq ($(filter dramsim3, $(subst _, ,$(CL_MANYCORE_MEM_CFG))),)
+CL_MANYCORE_MEMSYS_FEATURE_DMA   := 1
+CL_MANYCORE_MEMSYS_FEATURE_CACHE := 1
+endif
+
+# do we use infinite memory?
+ifneq ($(filter infinite, $(subst _, ,$(CL_MANYCORE_MEM_CFG))),)
+CL_MANYCORE_MEMSYS_FEATURE_DMA   := 1
+CL_MANYCORE_MEMSYS_FEATURE_CACHE := 0
+endif
+
```

Comment:
Thanks! 

I think there's still a discussion to be had about where we put memory-specific definitions but for now this is great!